apps/src/lab2/redux/lab2ReduxSelectors.ts

```diff
@@ -44,50 +45,59 @@ export const isReadOnlyWorkspace = (state: RootState) => {
   } else if (isViewingExemplar) {
     return true;
   }
-  // Otherwise, we are in read only mode if we are not the owner of the channel,
-  // the level is frozen, the level is a read only predict level, the level has been submitted.
-  // or this is a lab that should be read only while running and the code is currently running.
+  // Otherwise, we are in permanently read-only mode if we are not the owner of the channel,
+  // the level is frozen, the level is a read-only predict level, the level has been submitted,
+  // viewing an old version, or in widget view.
   const isOwner = state.lab.channel?.isOwner;
   const isFrozen = !!state.lab.channel?.frozen;
-  const readonlyPredictLevel = isReadonlyPredictLevel(state);
+  const readOnlyPredictLevel = isReadOnlyPredictLevel(state);
   const hasSubmitted = getCurrentLevel(state)?.status === LevelStatus.submitted;
   const isViewingOldVersion = state.lab2Project.viewingOldVersion;
-  const isRunningAndReadonly =
-    (state.lab2System.isRunning || state.lab2System.isValidating) &&
-    shouldBeReadonlyWhileRunning(state);
 
   return (
     !isOwner ||
     isFrozen ||
-    readonlyPredictLevel ||
+    readOnlyPredictLevel ||
     hasSubmitted ||
-    isRunningAndReadonly ||
     isViewingOldVersion ||
```

Comment:
If they are viewing an old version we want to count that as "permanent"?

Reply:
Good point. Can move this check to `isReadOnlyWorkspace`. I think `hasSubmitted` is okay to stay here though. Lmk what you think.

Reply:
Okay, I think `hasSubmitted` should move to `isReadOnlyWorkspace` (discussed on Slack) because this selector is used to determine if we show version history tab/panel or not for `lab2` labs.